src/Common/src/TypeSystem/IL/ILProvider.cs

```diff
@@ -101,9 +101,10 @@ private MethodIL CreateMethodIL(MethodDesc method)
                 if (method.IsPInvoke)
                 {
                     var pregenerated = McgInteropSupport.TryGetPregeneratedPInvoke(method);
-                    if (pregenerated == null)
-                        return PInvokeMarshallingILEmitter.EmitIL(method);
-                    method = pregenerated;
+                    if (pregenerated != null)
+                        method = pregenerated;
```

Comment:
The pre-generated methods should be taken as-is (they are basically what `PInvokeMarshallingILEmitter` produces).